src/common/points.rs

```diff
@@ -226,6 +257,88 @@ pub async fn do_clear_payload(
     .await
 }
 
+pub async fn do_batch_update_points(
+    toc: &TableOfContent,
+    collection_name: &str,
+    operations: Vec<UpdateOperation>,
+    shard_selection: Option<ShardId>,
+    wait: bool,
+    ordering: WriteOrdering,
+) -> Result<Vec<UpdateResult>, StorageError> {
+    try_join_all(operations.into_iter().map(|operation| async {
```

Comment:
yeah, ordering is definitely a concern here. 

IMO there are several possible solutions:
 - Run the updates sequentially, waiting for each to complete before starting the next one
 - Fire off all updates sequentially through the same node and then await them all in parallel
 - Group the update operations into a single internal message to the nodes (best guarantees of ordering and most optimization potential, but least separation of concerns and most work to plumb through)

WDYT?

Reply:
> Run the updates sequentially, waiting for each to complete before starting the next one

I'm _afraid_ it should be like this. I'm not too sure about the others.

Maybe we can add a flag to disable strict ordering for this. Then a user could decide that it may not be important for some non-overlapping operations. I'm not sure how @generall thinks about this.

Reply:
I changed it to what I _think_ should be enough to ensure ordering within a request. The operations might still be interleaved with update operations from other requests though. I'm not sure what the atomicity guarantees should be there.

Reply:
I believe your approach is correct. The propagated `ordering` flag provides extra guarantees.

Reply:
I believe your approach is correct. The propagated `ordering` flag provides extra guarantees.